internal/loader/synthetic_device_loader.go

```diff
@@ -55,3 +64,36 @@ func (sd *SyntheticDeviceLoader) BatchGetSyntheticDeviceByVehicleID(ctx context.
 
 	return results
 }
+
+// BatchGetSyntheticDeviceByVehicleID implements the dataloader for finding synthetic devices by their ids and returns
+// them in the order requested
+func (sd *SyntheticDeviceLoader) BatchGetSyntheticDeviceByID(ctx context.Context, ids []int) []*dataloader.Result[*gmodel.SyntheticDevice] {
+	results := make([]*dataloader.Result[*gmodel.SyntheticDevice], len(ids))
+
+	devices, err := models.SyntheticDevices(models.SyntheticDeviceWhere.ID.IN(ids)).All(ctx, sd.db.DBS().Reader)
+	if err != nil {
+		for i := range ids {
+			results[i] = &dataloader.Result[*gmodel.SyntheticDevice]{Data: nil, Error: err}
+		}
+		return results
+	}
+
+	sdByVehicleID := map[int]*models.SyntheticDevice{}
```

Comment:
Similarly to aftermarket devices, I don't understand the focus on vehicle ids here.